app/src/main/java/com/keylesspalace/tusky/ComposeActivity.java

```diff
@@ -1121,11 +1130,24 @@ private void makeCaptionDialog(QueuedMedia item) {
         DisplayMetrics displayMetrics = new DisplayMetrics();
         getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
 
-        Picasso.with(this)
-                .load(item.uri)
-                .resize(displayMetrics.widthPixels, displayMetrics.heightPixels)
-                .onlyScaleDown()
-                .into(imageView);
+        Single.fromCallable(() ->
+                MediaUtils.getSampledBitmap(getContentResolver(), item.uri, displayMetrics.widthPixels, displayMetrics.heightPixels))
+                .subscribeOn(Schedulers.computation())
+                .observeOn(AndroidSchedulers.mainThread())
+                .as(autoDisposable(from(this, Lifecycle.Event.ON_DESTROY)))
+                .subscribe(new SingleObserver<Bitmap>() {
+                    @Override
+                    public void onSubscribe(Disposable d) {}
+
+                    @Override
+                    public void onSuccess(Bitmap bitmap) {
+                        imageView.setImageBitmap(bitmap);
+                    }
+
+                    @Override
+                    public void onError(Throwable e) { }
```

Comment:
Did you use `SingleObserver` and not consumer to swallow errors? You could still use `BiConsumer` or two `Consumer`s to save a little bit of space but... idk. I just don't like Java's huge declarations but that's not important.